### ISSUE TITLE
Softmax applied twice in SSLOnlineEvaluator callback

### ISSUE THREAD

**Issue by norabelrose** [PyTorchLightning/lightning-bolts#662]
## 🐛 Bug

It looks like the `SSLEvaluator` module includes a `nn.Softmax()` operation as its last layer, but the `SSLOnlineEvaluator` callback, which trains an `SSLEvaluator` module, uses `F.cross_entropy_loss` as the loss function. This is mathematically incorrect and could lead to systematically worse prediction performance, since `F.cross_entropy_loss` expects raw, unnormalized scores instead of probabilities.

It seems like this could be fixed by simply removing the `nn.Softmax()` module from `SSLEvaluator`, as long as nothing else depends on that being there, or (somewhat less ideally) just taking the log of the probabilities from `SSLEvaluator` and then using `F.nll_loss`.

UPDATE: The bug also seems to appear in `SSLFinetuner`. It looks like removing the `nn.Softmax()` module from `SSLEvaluator` should fix the bug, although we'll have to make sure that the logits get softmax applied to them before being fed into any metrics.

### To Reproduce

Look at [evaluator.py](https://github.com/PyTorchLightning/lightning-bolts/blob/master/pl_bolts/models/self_supervised/evaluator.py) and then look at [ssl_online.py](https://github.com/PyTorchLightning/lightning-bolts/blob/master/pl_bolts/callbacks/ssl_online.py).
